dist/tools/riotboot_gen_hdr/Makefile

```diff
@@ -0,0 +1,39 @@
+RIOTBASE       := ../../..
+RIOT_INCLUDE   := $(RIOTBASE)/sys/include
+RIOT_CORE_INC  := $(RIOTBASE)/core/include
+NATIVE_INCLUDE := $(RIOTBASE)/cpu/native/include
+COMMON_SRC     := common.c
+COMMON_HDR     := common.h
+
+RIOT_HDR_SRC := \
+	$(RIOTBASE)/sys/checksum/fletcher32.c \
```

Comment:
Is it correct to peek into the contents of a module? I feel this violates module encapsulation.

Reply:
The compilation of the tools is a complete separated process, actually in a complete different environment. This line only makes (re)use of the C file, which has no dependencies so it's just like copy/pase it.